app/assets/javascripts/visualizations/highvis/baseVis.coffee.erb

```diff
@@ -113,279 +115,258 @@ $ ->
         gData
 
       ###
-      Draws y axis controls
-      This includes a series of checkboxes or radio buttons for selecting
-      the active y axis field(s).
+      A wrapper for making y-axis controls
+      See drawAxisControls for more detail
       ###
-      drawYAxisControls: (radio = false, enableDataPoint = false) ->
-        # Populate choices
-        yFields =
-          if enableDataPoint then data.normalFields
-          else data.normalFields[1..data.normalFields.length]
+      drawYAxisControls: (title, selFields, allFields, radio, iniRadio = 0,
+        handler = ->) ->
+        @drawAxisControls('y-axis', title, selFields, allFields, radio,
+          iniRadio, handler)
 
-        checked = false
-        controls = '<div id="yAxisControl" class="vis_controls">'
+        # Initialize and track the status of this control panel
+        globals.configs.yAxisOpen ?= false
+        initCtrlPanel('y-axis-ctrls', 'yAxisOpen')
 
+      ###
+      Draws axis controls
+      This includes a series of checkboxes or radio buttons for selecting
+      the active axis field(s).
+        id         Identifier prefix for the control panel
+        title      Title for the control panel
+        selFields  Object that tracks the current field selection
+        allFields  Listing of valid fields
+        radio      Boolean to draw radio buttons instead of checkboxes
+        iniRadio   Initial value of radio if radio == true (optional)
+        handler    A function of two parameters to handle selection events
+                   Takes the form: (selected, selFields) ->
+      ###
+      drawAxisControls: (id, title, selFields, allFields, radio, iniRadio = 0,
+        handler = ->) ->
+        # If there is only one option, then no radio button interaction is
+        # possible
+        if radio and allFields.length <= 1 then return
+
+        # Configure the y-axis controls
+        ictx = {}
+        ictx.fields = for f in allFields
+          id:    id + '-field-' + f
+          label: fieldTitle(data.fields[f])
+          value: f
+          logId: id + '-field-' + f
+          name:  id
+        template =  if radio then 'axis-radio' else 'axis-check'
+        unless radio
+          ictx.checkall =
+            id:    'ckall-' + id
+            label: '#Select All'
+            logId: 'check-all-' + id
+
+        # Draw the y-axis controls
+        octx = {}
+        octx.id = id + '-ctrls'
+        octx.title = title
+        octx.body = HandlebarsTemplates[hbCtrl(template)](ictx)
+        axisTools = HandlebarsTemplates[hbCtrl('body')](octx)
+        $('#vis-ctrls').append axisTools
+
+        # Initialize checked and make y axis checkbox/radio handler
         if radio
```

Comment:
inctx and outctx is the best  you're getting :stuck_out_tongue_closed_eyes: Mike and I agreed that this was sufficient.